.github/workflows/build.yml

```diff
@@ -0,0 +1,32 @@
+name: Build
+
+on:
+  pull_request:
+  push:
+    branches:
+      - main
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
```

Comment:
Interesting! I'm almost disappointed in Github with how limited configuration is allowed here. Gitlab allows you to specify a full docker name.

Reply:
If we wanted to we could run the whole shebang in a container of our choice. The runs-on command controls which VM type it runs on. At work we have like 4 for various configurations.

_Within_ that VM you can then control what the job actually executes inside of. Here it'll run on the bare machine, but [we can run it in a container if we want to](https://docs.github.com/en/actions/learn-github-actions/workflow-syntax-for-github-actions#jobsjob_idcontainer) instead. The workflow syntax is fairly powerful in my experience, especially when you start laying into the conditional step systems.